src/axom/sidre/tests/spio/spio_serial.hpp

```diff
@@ -176,3 +176,49 @@ TEST(spio_serial, write_read_write)
   IOManager writer_b(MPI_COMM_WORLD);
   writer_b.write(ds_r.getRoot(), num_files, filename, PROTOCOL);
 }
+
+//------------------------------------------------------------------------------
+TEST(spio_serial, rootfile_suffix)
+{
+  // This test verifies the usage of the string passed to writer.write() to
+  // control the names of the root file and the output files. The usual
+  // case is that the suffix ".root" is appended to the string to name the
+  // root file, and the given string is used to name the subdirectory holding
+  // the data files. The exception is if the string already has the suffix
+  // ".root", is is used unchanged to name the root file, and the suffix is
+  // removed to create a base string to name the subdirectory and the data
+  // files.
+
+  int my_rank, num_ranks;
+  MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
+  MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
+
+  int num_files = num_ranks;
+
+  std::string suffix_name = "hassuffix" + ROOT_EXT;
+  std::string nosuffix_name = "nosuffix";
+
+  // Initialize a datastore and dump to disk
+  DataStore* ds = new DataStore();
+  ds->getRoot()->createViewScalar<int>("grp/i", 2);
+  IOManager writer(MPI_COMM_WORLD);
+
+  // Write output to two file locations, with and without root suffix
+  writer.write(ds->getRoot(), num_files, suffix_name, "sidre_hdf5");
+  writer.write(ds->getRoot(), num_files, nosuffix_name, "sidre_hdf5");
+
+  // Create two more DataStores to read in the both outputs
+  DataStore ds_suffix, ds_nosuffix;
+  IOManager reader(MPI_COMM_WORLD);
+  reader.read(ds_suffix.getRoot(), suffix_name);
+  reader.read(ds_nosuffix.getRoot(), nosuffix_name + ROOT_EXT);
```

Comment:
Clarification: It looks like `IOManager.read()` still needs the `.root` suffix. 
Is that correct?

Reply:
Strictly speaking, no. `write()` always produces a file with the `.root` suffix, so in conventional usage, that filename should be provide unchanged to `read()`. But someone could rename the file to anything else and provide the new filename to `read()`, and this should still work, as long at the file's contents remain valid.

Reply:
Thanks -- got it -- so, `read()` does not alter the path to append/remove a suffix, but `write()` does